tt_metal/host_api.hpp

```diff
@@ -318,7 +319,7 @@ void EnqueueWriteBuffer(CommandQueue& cq, Buffer& buffer, const void* src, bool
  * | program      | The program that will be executed on the device that cq is bound to    | Program &                     |                                    | Yes      |
  * | blocking     | Whether or not this is a blocking operation                            | bool                          |                                    | Yes      |
  */
-void EnqueueProgram(CommandQueue& cq, Program& program, bool blocking);
+void EnqueueProgram(CommandQueue& cq, Program& program, bool blocking, Trace* trace = nullptr);
```

Comment:
I noticed that you use `Trace &` below, but `Trace *` here. Can we use `std::optional < std::reference_wrapper<Trace> > ` instead?